model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/LdapMapStorageProvider.java

```diff
@@ -39,13 +40,18 @@ public void close() {
     @Override
     @SuppressWarnings("unchecked")
     public <V extends AbstractEntity, M> MapStorage<V, M> getStorage(Class<M> modelType, Flag... flags) {
-        return session -> {
-            MapKeycloakTransaction<V, M> sessionTx = session.getAttribute(sessionTxPrefix + modelType.hashCode(), MapKeycloakTransaction.class);
-            if (sessionTx == null) {
-                sessionTx = factory.createTransaction(session, modelType);
-                session.setAttribute(sessionTxPrefix + modelType.hashCode(), sessionTx);
+        // MapStorage is not a functional interface, therefore don't try to convert it to a lambda as additional methods might be added in the future
+        //noinspection Convert2Lambda
+        return new MapStorage<V, M>() {
+            @Override
+            public MapKeycloakTransaction<V, M> createTransaction(KeycloakSession session) {
+                MapKeycloakTransaction<V, M> sessionTx = session.getAttribute(sessionTxPrefix + modelType.hashCode(), MapKeycloakTransaction.class);
+                if (sessionTx == null) {
+                    sessionTx = factory.createTransaction(session, modelType);
+                    session.setAttribute(sessionTxPrefix + modelType.hashCode(), sessionTx);
+                }
```

Comment:
Should this be under `synchronized` guard, like e.g. here? https://github.com/keycloak/keycloak/blob/b356618cc2e00ef3ae37f515c1d4b3bdbcde3f62/model/jpa/src/main/java/org/keycloak/connections/jpa/DefaultJpaConnectionProviderFactory.java#L160-L162

Reply:
There is no need to a guard: The returned MapStorage is stateless and lightweight. The only method being called is called with a KeycloakSession as parameter, and all calls within a KeycloakSession are non-concurrent.

Reply:
There is no need to a guard: The returned MapStorage is stateless and lightweight. The only method being called is called with a KeycloakSession as parameter, and all calls within a KeycloakSession are non-concurrent. 
@hmlnarik

Reply:
Same argument appiles here as in https://github.com/keycloak/keycloak/pull/11328/files#r856132221, resolving